### DISCUSSION TITLE
How to specify VSCode (default) settings at code-server launch

### DISCUSSION THREAD

**Discussion by unode** [coder/code-server#4593] [Q&A]
Hi everyone,

Is there any way to customize default VSCode settings when code-server is launched?

Initially I thought I could use `code-server --config config.json` but this config should be YAML and affects only code-server itself not the underlying VSCode.

I'm trying to setup a JupyterHub VSCode using [dirkcgrunwald/jupyter_codeserver_proxy](https://github.com/dirkcgrunwald/jupyter_codeserver_proxy-) and I would like to have some settings using different defaults.

For instance:

```json
{
    "terminal.integrated.gpuAcceleration": "canvas"
}
```

to avoid some terminal rendering issues.

Is there any way to do this without forcing the setting into the user's `~/.local/share/code-server/User/settings.json` file?
We'd like to set these as defaults but still allow the user to override it if they modify their settings.

Thanks!

**Comment by lg-nineoneninethreenine** [comment]
Hmm... Have you tried going to the `config.json` in code-server? Hit the F1 key, type Settings, and hit the JSON mode.
Remember, some features are not integrated yet, so it might be a breaking change. If you don't want to break your installation, you could try using the `.vscode/config.json` method in your project root directory, always works.

**Reply by jsjoeio** [reply]
I'm not sure how https://github.com/dirkcgrunwald/jupyter_codeserver_proxy- works but it looks like there is a `.vscode/settings.json` in that directory. What happens if you add your custom settings there? Does that work?

**Comment by consideRatio** [comment]
System-wide default settings are tracked by https://github.com/microsoft/vscode/issues/27972. There is as of very recently system-wide mandatory settings though, which are called Policy settings, see https://code.visualstudio.com/docs/configure/settings#_settings-precedence.

I was on the lookout for system-wide default settings specifically, which still isn't around.
